libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/MyTurnContext.java

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.ai.qna;
+
+import com.microsoft.bot.builder.BotAdapter;
+import com.microsoft.bot.builder.DeleteActivityHandler;
+import com.microsoft.bot.builder.SendActivitiesHandler;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.builder.TurnContextStateCollection;
+import com.microsoft.bot.builder.UpdateActivityHandler;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.ConversationReference;
+import com.microsoft.bot.schema.InputHints;
+import com.microsoft.bot.schema.ResourceResponse;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public class MyTurnContext implements TurnContext {
+
+    private BotAdapter adapter;
+    private Activity activity;
+
+    public MyTurnContext(BotAdapter withAdapter, Activity withActivity) {
+        this.adapter = withAdapter;
+        this.activity = withActivity;
+    }
+
+    public String getLocale() {
```

Comment:
these get and set locale I didn't see them in the [C# file](https://github.com/microsoft/botbuilder-dotnet/blob/main/tests/Microsoft.Bot.Builder.AI.QnA.Tests/MyTurnContext.cs), and I see they throw non implemented exceptions. Is it some inheritance thing?

Reply:
That is because the Interface that are using. In C# it use [ITurnContext](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder/ITurnContext.cs) and in Java [TurnContext](https://github.com/microsoft/botbuilder-java/blob/main/libraries/bot-builder/src/main/java/com/microsoft/bot/builder/TurnContext.java). The interface in Java has the get and set for locale, but C# doesn't have it, so If we are going to implement the TurnContext Interface in Java we need to add them. And I'm throwing the non implemented exception to have consistency.